README.rst

```diff
@@ -26,3 +26,371 @@ Run `tox` to run the whole test suite:
       lint: commands succeeded
       flake8: commands succeeded
       congratulations :)
+
+Usage
+-----
+
+RIOTCtrl provides a python object abstraction of a RIOT device. It’s
+meant as a starting point for any serial abstraction on which higher
+level abstractions (like a shell) can be built.
+
+.. code:: python
+
+   from riotctrl.ctrl import RIOTCtrl
+
+   env = {'BOARD': 'native'}
+   # if not running from the application directory the a path must be provided
+   ctrl = RIOTCtrl(env=env, application_directory='.')
+   # flash the application
+   ctrl.make_run(['flash'])
+   # run the terminal through a contextmanager
+   with ctrl.run_term():
+       ctrl.term.expect('>')       # wait for shell to start
+       ctrl.term.sendline("help")  # send the help command
+       ctrl.term.expect('>')       # wait for the command result to finnish
+       print(ctrl.term.before)     # print the command result
+   # run without a contextmanager
+   ctrl.start_term()               # start a serial terminal
+   ctrl.term.sendline("help")      # send the help command
+   ctrl.term.expect('>')           # wait for the command result to finnish
+   print(ctrl.term.before)         # print the command result
+   ctrl.stop_term()                # close the terminal
+
+Creating a RIOTCtrl object is done via environments. If empty then all
+configuration will come from the target application makefile. But any
+Make environment variable can be overridden, for example setting
+``BOARD`` to a target ``BOARD`` which is not the default for that
+application.
+
+Any make target used on RIOT devices can be used on the abstraction
+like: ``make flash`` => ``ctrl.make_run(['flash'])``.
```

Comment:
"on the abstraction like"? I think there is a word missing here.

Reply:
Since this is the first mention of `make_run` I wouldn't replace it here, but maybe also mention `ctrl.flash()` here. Something like e.g.

```suggestion
Any make target used on RIOT devices can be used on the abstraction
like: ``make flash`` => ``ctrl.make_run(['flash'])``. Some make
targets are even further abstracted, e.g. ``ctrl.make_run(['flash'])``
can also be executed using the ``ctrl.flash()`` method.
```